src/link.pyx

```diff
@@ -136,6 +136,20 @@ cdef class cLink(StructBase):
             self._value_error()
         return value
 
+    @property
+    def desired_properties(self):
+        cdef c_amqpvalue.AMQP_VALUE value
+        if c_link.link_get_desired_capabilities(self._c_value, &value) != 0:
+            self._value_error()
+        return value
+
+    @property
+    def offered_properties(self):
+        cdef c_amqpvalue.AMQP_VALUE value
+        if c_link.link_get_offered_capabilities(self._c_value, &value) != 0:
+            self._value_error()
+        return value
```

Comment:
Do we need to call `value_factory` of the output to convert it to Python data?

Reply:
will update this